expect_test.go

```diff
@@ -341,6 +341,99 @@ func TestExpect_Branches(t *testing.T) {
 	e5.chain.assertFlags(t, 0)
 }
 
+func TestExpect_ErrorPropagation(t *testing.T) {
+	t.Run("subsequent operations", func(t *testing.T) {
+		ctr := 0
+		reporter := newMockReporter(t)
+		reporter.reportCb = func() {
+			ctr++
+		}
+
+		arr := NewArray(reporter, []interface{}{"foo"})
+		arr.IsEmpty().IsEmpty() // The second .IsEmpty() call does not report failure
```

Comment:
Here and in all other tests: could you rework tests to check flags and counter of all values after every operation? And also maybe add some comments.

In this case, it would be like:

```go
// failing operation
arr.IsEmpty()
arr.chain.assertFlags(...)
assert.Equal(t, 1, ctr) // reported

// subsequent failing operation on the same object
arr.IsEmpty()
arr.chain.assertFlags(...)
assert.Equal(t, 1, ctr) // second failure was not reported
``` 

(I think we can remove checks `assert.Equal(t, true, reporter.reported)` because we anyway check counter)

There are two benefits of this approach:

* tests are much more readable (though also more bloated), because by looking at the test you can understand what is going on at each step; with current test, you have to guess

* we can be sure that the counter or flag was set by exact desired operations, not some other, because we do a check after each operation

Reply:
Ok, I am agree with this. I will do the update on this PR.

Thanks!